### PR TITLE
MRG: Extract blox from algorithms

### LINKED ISSUE #66: Consolidate replay buffer implementations
Currently several algorithms use their own implementations of replay buffers resulting in inconsistencies and duplicate code. We need to create a single implementation of replay buffers to be reused in all algorithms. Flashbax might be an option if we want to use an external library to build upon.

### PR DESCRIPTION
**Pull request by AlexanderFabisch**
Fix #66 

* Extract `target_net.soft_target_net_update` from DDPG
* #66
    * Extract `replay_buffer.ReplayBuffer` from DDPG
    * Use it for DQN
    * Remove other replay buffer implementations
* Extract `losses.mse_continuous_action_value_loss` from DDPG
* Extract `losses.mse_discrete_action_value_loss` from DQN
* Extract `losses.mse_value_loss` from REINFORCE
* Extract `losses.stochastic_policy_gradient_pseudo_loss` from REINFORCE
* Extract `losses.deterministic_policy_gradient_loss` from DDPG
* Remove unused module `blox.value_functions_jax`
* Unify documentation of DDPG/TD3/SAC
* Add logger to DQN
* **Fix** DQN loss with `jax.lax.stop_gradient`
* **Fix** scaling of exploration noise in DDPG and TD3 (use `normal`, not `multivariate_normal`)

### DQN

Learning of DQN curves before: purple / green; after: blue

Screenshot_20250518_195154

(Still unstable, but I guess this will be fixed with target nets.)

### DDPG/TD3

New result on Hopper environment with DDPG/TD3:

* TD3: **3594.9609 +- 57.6224** with learning rate `1e-3`, before: 2881.7565 +- 863.0822 (see #101)
* DDPG: **2009.0401 +- 869.9238** with learning rate `1e-3`, before: 1446.0263 +- 425.4875 (see #101)

Examplary smoothed learninv curve with TD3:

metrics-18_09_32-23-May-25

New results on Pendulum:

* TD3: **-136.4113 +- 6.8865** with learning rate `1e-3`, before -151.1679 +- 48.6921 (see #101) 
* DDPG: **-135.4101 +- 0.8654** with learning rate `1e-3`, before -135.4195 +- 6.5115 (see #101)

### CODE REVIEW

File: rl_blox/blox/replay_buffer.py [hunk 1]
```diff
@@ -1,57 +1,103 @@
...
+    insert_idx: int
 
-class ReplayBufferJax:
-    buffer: deque[Tuple[ArrayLike, ArrayLike, float, ArrayLike, bool]]
+    def __init__(
+        self,
+        buffer_size: int,
+        keys: list[str] | None = None,
+        dtypes: list[npt.DTypeLike] | None = None,
+        discrete_actions: bool = False,
```
[the remark this answers is not in the document]
Comment (AlexanderFabisch) [reply rl_blox/blox/replay_buffer.py:45]: Yes, that's future work!
